PYTHON_CODING.md

```diff
@@ -0,0 +1,216 @@
+
+- [reStructuredText](http://docutils.sourceforge.net/docs/user/rst/quickstart.html): The most common format. Very flexible and natively supported by Sphinx.
+- [Numpydoc](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt): Generally more readable, but harder to comply to. A plugin adds Numpydoc support to Sphinx.
+- Google style: The Numpydoc syntax is based on this.
+- Epydoc: Based on Javadoc. It is now discontinued (last release was 2008), but it can still be found in few projects.
+
+See this [tutorial on documentation]() to get started. Here is a very minimal example with Numpydoc syntax. If you have a module called `mymodule.py`
+```python
+"""Docstring for my module.py.
+
+All docstrings start and end with triple double quotes.
+
+"""
+
+
+def add_numbers(number1, number2):
+    """For obvious functions a single-line description is enough."""
+    return number1 + number2
+
+
+class MyClass:
+    """One-line summary of MyClass.
+
+    For more complex code, give a one-line summary followed by a
+    further description that can go over multiple lines.
+
+    Attributes
+    ----------
+    my_float : float
+        Description of attribute.
+
+    """
+
+    def __init__(self, my_float):
+        self.my_float = my_float
+
+    def multiply(self, number):
+        """Return number * self.my_float.
+
+        Parameters
+        ----------
+        number : float
+            This is multiplied by self.my_float.
+
+        Returns
+        -------
+        float
+            Result of multiplication.
+
+        """
+        return self.my_float * number
+```
+you can use
+```
+python
+>>> import mymodule
+>>> help(mymodule)
+>>> help(mymodule.MyClass.multiply)
+```
+to obtain information on how to use the code in `mymodule.py`.
+
+
+## Find bugs before they find you
+Nothing wastes your time like a bug that does not raise an error. You can spend several weeks on some results before you realize your code is incorrect. It is even worse when you realize it after publication. There are few practices that can help you to **prevent introducing bugs.**
+- **First and foremost, write tests!** The paradigm of test-driven development is simple:
+  1. Create a test that fails, so that you know you are testing the right thing.
+  2. Implement only the code that makes you pass the test.
+  3. Clean up your code (refactor, update docs, etc.) and repeat.
+
+  Several Python tools (pytest, nose2, unittest), allow you to easily run tests automatically. This will lower the chances of introducing regressions when you modify your code. See this [tutorial on testing]() for more information, and to get started.
```

Comment:
Can you link to [`pytest`](http://doc.pytest.org/en/latest/), [`nose2`](http://nose2.readthedocs.io/), and [`unittest`](https://docs.python.org/3/library/unittest.html)?